### PR TITLE
Allow Mappings to be given as the options in a selection widget again.

### PR DESCRIPTION
**Pull request by jasongrout**
We deprecated this long ago when dicts were not ordered (#2130, #1958, #2679), then removed it in ipywidgets 8.0. However, now dicts are ordered by default, so perhaps we can put it back in and keep people's code from breaking.

I want to place a high value on easy ways to support backwards compatibility, as many people would rather just not change their code if they can help it.

However, if we are going to stick with this decision in 8.0, #3556 makes the error more helpful. #3556 and this PR are mutually exclusive - at most one should be merged.

### CODE REVIEW

**Comment by jasongrout** [comment]
The expected tests fail. If we decide to do this PR, I can fix the tests.

**Review by vidartf — APPROVED** [review]
There was a general consensus in the dev meeting today that it was probably best to add back the support now that dicts are ordered by default, but to not put any use/mention of this pattern in the docs. @maartenbreddels also argued for not mentioning it in the docstring either, but instead just leave a comment in the code that this support is here for backwards compatibility reasons.

**Comment by vidartf** [comment]
Tests will then need fixing.

**Comment by jasongrout** [comment]
@vidartf - rerequesting review, since tests pass now.

Also pinging @jtpio and @pbugnion, who helped remove the dict option and deprecated it in 7.x (though it appears the deprecation warning was never really seen by users?)

**Comment by jtpio** [comment]
> though it appears the deprecation warning was never really seen by users?

I believe the warning was shown as suggested by the second screenshot in https://github.com/jupyter-widgets/ipywidgets/pull/2679#issuecomment-577199266?

**Review by vidartf — APPROVED** [review]

File: python/ipywidgets/ipywidgets/widgets/tests/test_interaction.py [hunk 1]
```diff
@@ -187,17 +187,33 @@ def test_dict():
...
+    c = interactive(f, lis=values)
+    assert len(c.children) == 2
+    d = dict(
+        cls=widgets.Dropdown,
+        value=first,
+        options=values,
+        _options_labels=("3", "1", "2"),
+        _options_values=(300, 100, 200),
+    )
+    check_widgets(c, lis=d)
```
Comment (vidartf) [inline python/ipywidgets/ipywidgets/widgets/tests/test_interaction.py:216]: The above check uses `**check` while this uses `lis=d`. I'm not sure if I understand the nuance of why that works, but can maybe look again on Tuesday.

**Comment by vidartf** [comment]
I'll leave off the merge for now to leave some time for others to comment, but looks ok to me!

**Comment by jasongrout** [comment]
> I believe the warning was shown as suggested by the second screenshot in [#2679 (comment)](https://github.com/jupyter-widgets/ipywidgets/pull/2679#issuecomment-577199266)?

Only if you reset the DeprecationWarning filter to always show deprecation warnings, as shown in that screenshot. I imagine a vanishingly small fraction of our users did this?

File: python/ipywidgets/ipywidgets/widgets/tests/test_interaction.py (hunk 1 above)
Comment (jasongrout) [reply python/ipywidgets/ipywidgets/widgets/tests/test_interaction.py:216]: Great question. One is using check_widget, the other is using check_widgets :). That got me for a bit too. I'll rename check_widgets to check_widget_children to be more clear.

**Comment by jtpio** [comment]
> Only if you reset the DeprecationWarning filter to always show deprecation warnings (or ran Python with the appropriate switches to always show deprecation warnings, etc.), as shown in that screenshot. I imagine a vanishingly small fraction of our users did this?

Oh right, then yes probably not many users were able to see it. 

In that case yes it makes sense to revert it since most users probably didn't get the notice :+1:

**Pull request merged**